### PR TITLE
K blocker rc

### PR DESCRIPTION
**Pull request by BruceK4t1qbit**
Changed the k-blocker observations from using a single integer to encode the position of each block to using 2 integers (the x and y coordinates) to encode the position of each block.

Changed tests accordingly, fixed one test, and created one new test for a bug I created.

### CODE REVIEW

File: gym/envs/nysaq/tests/assets/rbmConfig1bt [hunk 1]
```diff
@@ -24,7 +24,7 @@
     "name": "bt",
     "num_blockers": 1,
     "grid_y": 5,
-    "positive_reward": 1,
-    "negative_reward": 0
+    "positive_reward APPARENTLY NOT USED": 1,
```
Comment (majiddadashi) [inline gym/envs/nysaq/tests/assets/rbmConfig1bt]: Can we actually use these instead? we can rename them:
win_reward: For winning. currently 5.
play_reward: For taking each action. currently 0.

File: gym/envs/nysaq/kblocker.py [hunk 2]
```diff
@@ -105,15 +107,19 @@ def _reset(self):
         # Initialize the agents randomly
         self.__coords = [(r, 0) for r in random.sample(range(self.__grid_x), self.__num_bs + 1)]
 
-        return np.array(list(map(self.__encode__, self.__coords)))
+        return self._get_encoded_state()
+
+
+    def _get_encoded_state(self):
+        return sum(map(list, self.__coords), [])
```
Comment (majiddadashi) [inline gym/envs/nysaq/kblocker.py]: Could you please add a comment here for readability? took me a few minutes to figure out what's happening here.

**Review by majiddadashi — CHANGES_REQUESTED** [review]

File: gym/envs/nysaq/tests/assets/rbmConfig1bt (hunk 1 above)
Comment (BruceK4t1qbit) [reply gym/envs/nysaq/tests/assets/rbmConfig1bt]: Implemented

File: gym/envs/nysaq/kblocker.py (hunk 2 above)
Comment (BruceK4t1qbit) [reply gym/envs/nysaq/kblocker.py]: Done.  And refactored the code so it is much simpler!

**Review by majiddadashi — APPROVED** [review]

**Pull request merged**